js/ui/overlay/overlay_drag.js

```diff
@@ -0,0 +1,164 @@
+
+        this._$container = dragContainer;
+        this._outsideMultiplayer = outsideMultiplayer;
+        this._$draggableElement = draggableElement;
+        this._updatePositionChangeHandled = updatePositionChangeHandled;
+
+        if(!dragEnabled) {
+            return;
+        }
+
+        eventsEngine.on(dragTarget, startEventName, (e) => { this._dragStartHandler(e); });
+        eventsEngine.on(dragTarget, updateEventName, (e) => { this._dragUpdateHandler(e); });
+    }
+
+    moveTo(top, left, e) {
+        e.preventDefault();
+        e.stopPropagation();
+
+        const allowedOffsets = this._allowedOffsets();
+        const offset = {
+            top: fitIntoRange(top, -allowedOffsets.top, allowedOffsets.bottom),
+            left: fitIntoRange(left, -allowedOffsets.left, allowedOffsets.right)
+        };
+        this._changePosition(offset);
+    }
+
+    _dragStartHandler(e) {
+        e.targetElements = [];
+
+        this._prevOffset = { x: 0, y: 0 };
+
+        const allowedOffsets = this._allowedOffsets();
+        e.maxTopOffset = allowedOffsets.top;
+        e.maxBottomOffset = allowedOffsets.bottom;
+        e.maxLeftOffset = allowedOffsets.left;
+        e.maxRightOffset = allowedOffsets.right;
+    }
+
+    _deltaSize() {
+        const $draggableElement = this._$draggableElement;
+        const $container = this._$container;
+
+        const contentWidth = $draggableElement.outerWidth();
+        const contentHeight = $draggableElement.outerHeight();
+        let containerWidth = $container.outerWidth();
+        let containerHeight = $container.outerHeight();
+        const document = domAdapter.getDocument();
+        if(this._isWindow($container)) {
+            const fullPageHeight = Math.max($(document).outerHeight(), containerHeight);
+            const fullPageWidth = Math.max($(document).outerWidth(), containerWidth);
+
+            containerHeight = fullPageHeight;
+            containerWidth = fullPageWidth;
+        }
+
+        const outsideMultiplayer = this.outsideMultiplayer() ?? 0;
+
+        return {
+            width: containerWidth - contentWidth + (contentWidth * outsideMultiplayer),
+            height: containerHeight - contentHeight + (contentHeight * outsideMultiplayer),
+        };
+    }
+
+    _dragUpdateHandler(e) {
+        const offset = e.offset;
+        const prevOffset = this._prevOffset;
+        const targetOffset = {
+            top: offset.y - prevOffset.y,
+            left: offset.x - prevOffset.x
+        };
+
+        this._changePosition(targetOffset);
+
+        this._prevOffset = offset;
+    }
+
+    _allowedOffsets() {
+        const overlayPosition = getOffset(this._$draggableElement.get(0));
+        const dragAreaPosition = this._isWindow(this._$container)
+            ? { top: 0, left: 0 }
+            : getOffset(this._$container.get(0));
+        const deltaSize = this._deltaSize();
+        const isAllowedDrag = deltaSize.height >= 0 && deltaSize.width >= 0;
+
```

Comment:
let's write here
```
if (isDragAllowed) {
    return {
       top: 0,
       bottom: 0,
       left: 0,
       right: 0
    }
}
```

to avoid excess calculations

Reply:
but we need these calculations later